src/Modules/UngradedModules.jl

```diff
@@ -118,11 +109,45 @@ function AbstractAlgebra.extra_name(F::FreeMod)
 end
 
 function show(io::IO, F::FreeMod)
-  @show_name(io, F)
-  @show_special(io, F)
+  compact = get(io, :compact, false)
```

Comment:
the 2 wierd show calls should stay.  show_name will print the variable name if assigned, show_special allows the override of the show function (free modules representing homogenous component stuff)

Reply:
Added it back in.